guide/sapphire/args.md

```diff
@@ -0,0 +1,147 @@
+
+  This is troublesome if you want to allow sending quotes within the content, as those would be excluded from `rest`, so `quotes` must be set to an empty array. This way, when `@bot echo Hello "there!"` is sent, your bot will reply with `Hello "there!"`.
+
+  ::: warning
+  `args.rest()` consumes all the parameters and then processes them, so you will not be able to consume any more parameters after it.
+  :::
+
+## Multiple arguments
+
+Let's make an `add` command, which takes two numbers and adds them. Create an `add.js` file in your `commands` folder and set up the command class again. Since two arguments are needed, use `args.pick()` twice:
+
+```js {9-11}
+module.exports = class AddCommand extends Command {
+	constructor(context) {
+		super(context, {
+			description: 'Adds two numbers',
+		});
+	}
+
+	async run(message, args) {
+		const a = await args.pick('number');
+		const b = await args.pick('number');
+		return message.channel.send(`The result is ${a + b}!`);
+	}
+};
+```
+
+Each time `args.pick('<type>')` is called, it consumes the arguments one by one and parses them as the `<type>` given. So `@bot add 1 2` would return `The result is 3!`.
+
+## Optional arguments
+
+Sapphire offers two options regarding regarding optional arguments. Let's make a `pow.js` command that takes two numbers, where the `base` is required, and the `exponent` defaults to 2.
+
+- **Optional Pattern**
+  ```js {5-6}
+  module.exports = class PowCommand extends Command {
+  	// ...
+  
+  	async run(message, args) {
+  		const base = await args.pick('number');
+  		const exponent = await args.pick('number').catch(() => 2);
+  		return message.channel.send(`The result is ${Math.pow(base, exponent)}!`);
+  	}
+  };
+  ```
+
+  Since `exponent` defaults to `2`, this allows users to send `@bot pow 4`, returning `The result is 16!`. However, this pattern ignores any possible errors, including invalid numbers, so `@bot pow 4 hello!` would return the same response as before. There is another pattern to solve this:
+
+- **Semi-required Pattern**
+  ```js {6}
+  module.exports = class PowCommand extends Command {
+  	// ...
+  
+  	async run(message, args) {
+  		const base = await args.pick('number');
+  		const exponent = args.finished ? 2 : await args.pick('number');
+  		return message.channel.send(`The result is ${Math.pow(base, exponent)}!`);
+  	}
+  };
+  ```
+
+  This is slightly more verbose, but it defaults `exponent` to `2` **only** if there were no parameters left to parse, so `@bot pow 4` will return `The result is 16!`, while `@bot pow 4 hello!` will return an error.
+
+Each pattern has its use-cases. The former is specially helpful for handling several optional parameters, although you can also discriminate all errors from `args` by checking `error.identifier`. This way you can tell if a number failed to parse for reasons such as it's not a number, it doesn't meet the minimum and/or maximum (if set), among other possible options.
```

Comment:
```suggestion
Each pattern has its use-cases. The former is especially helpful for handling several optional parameters, although you can also discriminate all errors from `args` by checking `error.identifier`. This way you can tell if a number failed to parse for reasons such as it's not a number, it doesn't meet the minimum and/or maximum (if set), among other possible options.
```
Yes, `specially` can be used in many different cases. However, it is being incorrectly used in this case, so it's better to use `especially`.